packages/pantheon/src/index.ts

```diff
@@ -0,0 +1,177 @@
+  const sideBoard = make(10).reduce((acc, _, idx) => {
+    const currentArchs = [chief.archetype, ...acc.map(({ archetype }) => archetype)];
+    const deity = createDeity(dispositions, currentArchs);
+
+    if (d(10) === 8) {
+      relationships.push({
+        first: d(2) === 1 ? 'anon-chronos' : 'anon-uranus',
+        second: deity.id,
+        action: 'beget',
+      });
+    }
+
+    acc.push(deity);
+    return acc;
+  }, []);
+
+  // Step 3. Determine if the chief should be matriarchal and self-procreating
+  // If the chief is a hermaphrodite, he will also be selfReproducing.
+  // Based on this, a spouse will either be created or the chief assigned
+  // as spouse.
+  const shouldBeMatriarch = chief.gender === 'female' && d(50) === 50;
+  const selfReproducing = shouldBeMatriarch || chief.gender === 'hermaphrodite';
+  const spouse = selfReproducing
+    ? { id: chief.id, gender: chief.gender, relationships: [] }
+    : createDeity(dispositions, []);
+
+  deities.push(...[chief, spouse, ...sideBoard]);
+
+  // Step 4. Generate consort relationships. These are randomly created, with the chief
+  // and chief's spouse being guaranteed. Anon are mortal (or animal) consorts
+  // that will produce demigods.
+  const consorts = [
+    {
+      first: chief.id,
+      second: spouse.id,
+      guarantee: 3,
+    },
+    ...make(8).map(() => ({
+      first: chief.id,
+      second: 'anon',
+    })),
+    ...(selfReproducing
+      ? []
+      : make(8).map(() => ({
+          first: spouse.id,
+          second: 'anon',
+        }))),
+  ];
+
+  const unpackGeneration = (acc, { relationships: generatedRelationships, children }) => {
+    if (!children.length) return acc;
+
+    relationships.push(...generatedRelationships);
+    acc.push(...children);
+    return acc;
+  };
+
+  // Step 5. Make babies
+  const secondGeneration = consorts
+    .map(
+      reproduce(
+        dispositions,
+        deities.map(({ archetype }) => archetype),
+      ),
+    )
+    .reduce(unpackGeneration, []);
+  deities.push(...secondGeneration);
+
+  const availableConsorts = deities.filter((d) => {
+    const hasTwoParents = relationships.filter(({ second }) => second === d.id);
+
+    return hasTwoParents.length === 2;
+  });
+
+  // step 6. create consorts for all non-demigods from gen-2 against
+  // all non-demigods in the pantheon.
+  const secondGenConsorts = secondGeneration.reduce((acc, deity) => {
+    if (d(2) !== 1) return acc;
+
+    availableConsorts.forEach((potentialSuitor) => {
+      if (potentialSuitor.id === deity.id) return;
+
+      if (d(3) === 1) {
+        acc.push({
+          first: deity.id,
+          second: potentialSuitor.id,
+          guarantee: d(2),
+        });
+      }
+    });
+
+    acc.push(
+      ...make(d(6)).map(() => ({
+        first: deity.id,
+        second: 'anon',
+      })),
+    );
+
+    return acc;
+  }, []);
+
+  deities.push(
+    ...secondGenConsorts
```

Comment:
i might not be understanding where secondGenConsorts is really pulling these, but I think it looks like it's pushing relationship events to the accumulator and returning those to this, instead of actual deities.

Reply:
Ohhh this might be a bug. Whoops

Reply:
This is actually correct, but I will clean it up for better readability.